### PR TITLE
Add specfile

### LINKED ISSUE #1: Add RPM build
- [x] Add an inmanta.spec file that builds `python3-inmanta`, which follows the current `<year>.<minor>[.<patch>]` versioning. The RPM should require the `python3-inmanta-core` RPM and pin its version, similar to the `inmanta-service-orchestrator` spec file. It should also install the Python `inmanta` package (the one from this repo, not the source package) in `/opt/inmanta` (again, the `inmanta-service-orchestrator` spec file could be used as an example).
- [ ] Add an `inmanta-oss-release` and corresponding pyp and rpm build jobs on Jenkins, preferably by making the current `irt/ci/inmanta-core/` builds config-driven pipelines so they work for `inmanta-core` as well as `inmanta-oss`.

### PR DESCRIPTION
**Pull request by bartv**
# Description

Add specfile and make file

# Self Check:

- [ ] Attached issue to pull request
- [ ] Changelog entry

### CODE REVIEW

**Comment by sanderr** [comment]
@bartv I don't think this should close #1 unless we create a new ticket to resolve its second task.

**Comment by bartv** [comment]
> @bartv I don't think this should close #1 unless we create a new ticket to resolve its second task.

ouch, I missed that one. Lets take a look what I can do there.

File: inmanta.spec [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+%install
+# Copy the inmanta venv to BUILD directory to work around ownership issue
+cp -r --no-preserve=ownership /opt/inmanta inmanta-venv
+chmod -x LICENSE
+
+site_packages_dir="inmanta-venv/lib/python3.6/site-packages"
+files=$(find "${site_packages_dir}" -maxdepth 1 -mindepth 1)
+
+# Install inmanta-license
```
Comment (arnaudsjs) [inline inmanta.spec:52]:
```suggestion
# Install inmanta
```

File: inmanta.spec [hunk 2]
```diff
@@ -0,0 +1,83 @@
+# Use release 0 for prerelease version.
+%define release 0
```
Comment (arnaudsjs) [inline inmanta.spec:2]:
```suggestion
%define release 1
```

File: inmanta.spec [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+Group:          Development/Languages
+License:        EULA
+URL:            http://inmanta.com
+Source0:        inmanta-%{sourceversion_egg}.tar.gz
+BuildRoot:      %{_tmppath}/%{name}-%{version}-%{release}-root-%(%{__id_u} -n)
+
+BuildRequires:  python3-inmanta-core
+BuildRequires:  systemd
+
+Requires:  python3-inmanta-core
```
Comment (arnaudsjs) [inline inmanta.spec:32]: Shouldn't we also add `python3-inmanta-agent` and `python3-inmanta-server`?

File: inmanta.spec [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+# Install inmanta-license
+%{_p3} -m pip install --no-index .
+
+# Only keep new packages
+rm -rf ${files}
+
+mkdir -p %{buildroot}/opt/inmanta/
+cp -r inmanta-venv/lib/ %{buildroot}/opt/inmanta/
+mkdir -p %{buildroot}/etc/inmanta/inmanta.d/
```
Comment (arnaudsjs) [inline inmanta.spec:60]: Why is this required?

File: inmanta.spec [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+# Remove other code from env
```
Comment (arnaudsjs) [inline inmanta.spec:62]: leftover comment?

File: inmanta.spec [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+%clean
+rm -rf %{buildroot}
+
+%files
+%defattr(-,root,root,-)
+%doc LICENSE
+/opt/inmanta/lib/
+/etc/inmanta
```
Comment (arnaudsjs) [inline inmanta.spec:71]: Why is this required?

**Review by arnaudsjs — CHANGES_REQUESTED** [review]

File: inmanta.spec (hunk 3 above)
Comment (bartv) [reply inmanta.spec:32]: no, because then you cannot install only core

File: inmanta.spec (hunk 4 above)
Comment (bartv) [reply inmanta.spec:60]: I kept it from the iso release. It is a bit strange that we do not have I in core actually.

File: inmanta.spec (hunk 5 above)
Comment (bartv) [reply inmanta.spec:62]: https://github.com/inmanta/inmanta-service-orchestrator/blob/master/inmanta.spec#L76

File: inmanta.spec (hunk 6 above)
Comment (bartv) [reply inmanta.spec:71]: I will remove these

File: inmanta.spec (hunk 2 above)
Comment (bartv) [reply inmanta.spec:2]: This is on master, so a prelease. That have to be 0

File: inmanta.spec (hunk 1 above)
Comment (bartv) [reply inmanta.spec]: fixed

File: inmanta.spec (hunk 4 above)
Comment (bartv) [reply inmanta.spec]: fixed

File: inmanta.spec (hunk 5 above)
Comment (bartv) [reply inmanta.spec]: fixed

File: inmanta.spec (hunk 6 above)
Comment (bartv) [reply inmanta.spec]: fixed

File: Makefile [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+	pip install -U -r requirements.txt
+	pip install -e .
+
+.PHONY: format
+format:
+	$(isort)
+	$(black)
+
+.PHONY: build-pytest-inmanta-extensions
+build-pytest-inmanta-extensions:
```
Comment (sanderr) [inline Makefile:50]: Is this target required for the `inmanta` repo?

**Review by sanderr — CHANGES_REQUESTED** [review]

File: inmanta.spec (hunk 2 above)
Comment (arnaudsjs) [reply inmanta.spec:2]: Note that the Makefile set this value to 0 when a non-stable release is requested. So it should be set to 1 by default and it's overridden for non-stable releases. I agree this is weird.

File: Makefile (hunk 7 above)
Comment (bartv) [reply Makefile:50]: no, probably not

Comment (bartv) [reply Makefile]: fixed

File: inmanta.spec (hunk 2 above)
Comment (bartv) [reply inmanta.spec:2]: ah, that is fine. But we might need to add a comment in the specfile :-)

**Review by arnaudsjs — CHANGES_REQUESTED** [review]

File: inmanta.spec (hunk 2 above)
Comment (bartv) [reply inmanta.spec]: damnit, I got this email but then afterwards I did not see the comment in the PR. Fix now (I hope)

**Review by arnaudsjs — APPROVED** [review]

**Review by sanderr — APPROVED** [review]
